src/app/modules/logs/logs.component.ts

```diff
@@ -0,0 +1,259 @@
+        this.modalSubscriptions.forEach((subscription: Subscription) => subscription.unsubscribe());
+    }
+
+    getLogs() {
+        this.logSubscription = this.logService.get().pipe(
+            finalize(
+                () => {
+                    this.loadingLogs = false;
+                }
+            )
+        ).subscribe(
+            response => {
+                if (response && response.length > 0) {
+                    this.logs = response.reduce((realLogs, res) => {
+                        const log = Log.apiToModel(res);
+                        if (log) {
+                            realLogs.push(log);
+                        }
+                        return realLogs;
+                    }, []);
+                    this.selectTab('distributions');
+                    this.createGraphs();
+                } else if (!response) {
+                    this.logs = null;
+                }
+            }
+        );
+    }
+
+    selectTab(tab: string): void {
+        this.selectedTab = tab;
+        if (this.logs) {
+            this.filteredLogList = this.logs.filter((log: Log) => log.get<string>('tabName') === tab);
+            this.filteredLogList.map((log: any) => {
+                if (!log.get('details')) {
+                    this.logService.getDetails(log);
+                }
+            });
+            this.logData = new MatTableDataSource(this.filteredLogList);
+            if (this.table && this.table.paginator) {
+                    this.table.paginator.pageIndex = 0;
+            } else if (this.tableMobile && this.tableMobile.paginator) {
+                    this.tableMobile.paginator.pageIndex = 0;
+            }
+        }
+    }
+
+    createGraphs() {
+        const sortedLogs = this.logs
+            .sort((log1: Log, log2: Log) => log1.get<Date>('date').getTime() - log2.get<Date>('date').getTime());
+        const oldestDate = sortedLogs[0].get<Date>('date').getFullYear();
+        const latestDate = sortedLogs[sortedLogs.length - 1].get<Date>('date').getFullYear();
+        const oldestDay = new Date();
+        oldestDay.setDate(oldestDay.getDate() - this.graphDuration);
+        const latestDay = new Date();
+        latestDay.setDate(latestDay.getDate() + 1);
+        const datePipe = new DatePipe('en-US');
+
+        const statusGraph = {
+            type: 'pie',
+            name: this.language.log_status_rate_title,
+            values: {}
+        };
+
+        const dayRequestGraph = {
+            type: 'line',
+            name: this.language.log_requests_day_title,
+            yLabel: this.language.log_requests,
+            xLabel: this.language.log_time,
+            values: {}
+        };
+
+        const activeUsersGraph = {
+            type: 'bar',
+            name: this.language.log_active_users_title,
+            xLabel: this.language.log_users,
+            yLabel: this.language.log_requests,
+            values: {}
+        };
+
+        this.requestsKHM = {
```

Comment:
Can't we have something general that will generate this value for all countries?